Headers/Public/VLCMediaList.h

```diff
@@ -104,12 +104,13 @@ extern NSString *const VLCMediaListItemDeleted;
 - (void)insertMedia:(VLCMedia *)media atIndex:(NSUInteger)index;
 
 /**
- * remove a media from a given position
+ * remove and return a media from a given position
  *
- * \param index the index of the media to remove
+ * \param index the index of the media to remove. index must be a valid index of the array.
+ * \return media at the given index
  * \note this function silently fails if the list is read-only
  */
-- (void)removeMediaAtIndex:(NSUInteger)index;
+- (VLCMedia *)removeMediaAtIndex:(NSUInteger)index;
```

Comment:
Replicates Swift STL's `func remove(at: Int) -> T `

Reply:
I don't understand why you want to return a VLCMedia instance that is gone from the list.

Reply:
Swift STL's `remove(at:)` behaves the same way and if the function doesn't returns `void`, how do I easily tell if a media had been deleted or not? A way to do is to raise NSRangeException, which leads to the use of `indexOf` without sanity checks in the function. Sorry if I'm not making sense 😅

Reply:
Considering BOOL as a return value to expose the successful deletion. This matches typical AppKit behavior.